From 8fe16559e38573fb47947d2c0658b1a83c2a6a89 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 13:42:53 +0200
Subject: [PATCH] [SW-0] pll: check all PLL div factors

Do not stop after first encountered disabled div factor.
Continue checking the others.

Change-Id: I719eaab5c8946e7da38ca584d2f6dc70f21ff37d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index dcaf956..28bad84 100644
--- a/pll.c
+++ b/pll.c
@@ -140,7 +140,7 @@
 		offset = sel * sizeof(u32);
 
 		if (cfg->div_factor[sel] == -1)
-			break;
+			continue;
 
 		if (bypass)
 			writel(0, cfg->base + offset + HL_PLL_DIV_SEL);
@@ -160,7 +160,7 @@
 		offset = sel * sizeof(u32);
 
 		if (cfg->div_factor[sel] == -1)
-			break;
+			continue;
 
 		writel(cfg->div_factor[sel], cfg->base +
 						offset + HL_PLL_DIV_FACTOR);
